spec/core/ics-003-connection-semantics/UPGRADES.md

```diff
@@ -0,0 +1,483 @@
+- The connection upgrade protocol should have the ability to change all connection-related parameters; however the connection upgrade protocol MUST NOT be able to change the underlying `ClientState`.
+The connection upgrade protocol MUST NOT modify the connection identifiers.
+
+## Technical Specification
+
+### Data Structures
+
+The `ConnectionState` and `ConnectionEnd` are defined in [ICS-3](./README.md), they are reproduced here for the reader's convenience. `UPGRADE_INIT`, `UPGRADE_TRY`, `UPGRADE_ERR` are additional states added to enable the upgrade feature.
+
+```typescript
+enum ConnectionState {
+  INIT,
+  TRYOPEN,
+  OPEN,
+  UPGRADE_INIT,
+  UPGRADE_TRY,
+}
+```
+
+- The chain that is proposing the upgrade should set the connection state from `OPEN` to `UPGRADE_INIT`
+- The counterparty chain that accepts the upgrade should set the connection state from `OPEN` to `UPGRADE_TRY`
+
+```typescript
+interface ConnectionEnd {
+  state: ConnectionState
+  counterpartyConnectionIdentifier: Identifier
+  counterpartyPrefix: CommitmentPrefix
+  clientIdentifier: Identifier
+  counterpartyClientIdentifier: Identifier
+  version: string | []string
+  delayPeriodTime: uint64
+  delayPeriodBlocks: uint64
+}
+```
+
+The desired property that the connection upgrade protocol MUST NOT modify the underlying clients or connection identifiers, means that only some fields of `ConnectionEnd` are upgradable by the upgrade protocol.
+
+- `state`: The state is specified by the handshake steps of the upgrade protocol.
+
+MAY BE MODIFIED:
+- `counterpartyPrefix`: The prefix MAY be modified in the upgrade protocol. The counterparty must accept the new proposed prefix value, or it must return an error during the upgrade handshake.
+- `version`: The version MAY be modified by the upgrade protocol. The same version negotiation that happens in the initial connection handshake can be employed for the upgrade handshake.
+- `delayPeriodTime`: The delay period MAY be modified by the upgrade protocol. The counterparty MUST accept the new proposed value or return an error during the upgrade handshake.
+- `delayPeriodBlocks`: The delay period MAY be modified by the upgrade protocol. The counterparty MUST accept the new proposed value or return an error during the upgrade handshake.
+
+MUST NOT BE MODIFIED:
+- `counterpartyConnectionIdentifier`: The counterparty connection identifier CAN NOT be modified by the upgrade protocol.
+- `clientIdentifier`: The client identifier CAN NOT be modified by the upgrade protocol
+- `counterpartyClientIdentifier`: The counterparty client identifier CAN NOT be modified by the upgrade protocol
+
+NOTE: If the upgrade adds any fields to the `ConnectionEnd` these are by default modifiable, and can be arbitrarily chosen by an Actor (e.g. chain governance) which has permission to initiate the upgrade.
+
+```typescript
+interface UpgradeTimeout {
+    timeoutHeight: Height
+    timeoutTimestamp: uint64
+}
+```
```

Comment:
I find the jump in concepts to be abrupt. Maybe you should use a subheader for each Data Structure?